### DISCUSSION TITLE
Example of GitHub Actions integration for testing

### DISCUSSION THREAD

**Discussion by jcfr** [Kitware/trame#46] [Q&A]
Ideally, I am looking into an example of workflow that would:
* start the application as a service in container
* have one or multiple test for:
   - triggering one or multiple action on the UI
   - performing a screen grab
* post the screen grab (or a collage of these) as a comment on the pull request. 

While the screenshot part and adding as a pull-request comment could be addressed using `swinton/screenshot-website`, `devicons/public-upload-to-imgur`, `actions/github-script` and `thollander/actions-comment-pull-request` (For example, see [this](https://github.com/Slicer/slicer.org/blob/0cd0fc7a75953f16fbd29f8ceb0ea212c012c337/.github/workflows/build-website.yml#L84-L136)), I am curious about first two steps.

**Comment by jourdain** [comment]
For the first step, the cookiecutter give you the frame for bundling your application into a docker image as a service. 
For the second, you can probably rely on some testing framework that rely on selenium for running and interacting with a browser.
All of that could be automated in python and maybe even on a ci.
